lib/membrane/rtp/demuxer/jitter_buffer.ex

```diff
@@ -0,0 +1,262 @@
+            RTP.PayloadFormat.resolve(
+              payload_type: packet.payload_type,
+              clock_rate: pad_options.clock_rate,
+              payload_type_mapping: payload_type_mapping
+            )
+
+          Process.send_after(
+            self(),
+            {:initial_latency_passed, packet.ssrc},
+            Membrane.Time.as_milliseconds(pad_options.jitter_buffer_latency, :round)
+          )
+
+          {clock_rate, pad_options.jitter_buffer_latency, true}
+      end
+
+    %State{
+      initial_latency_waiting: initial_latency_waiting,
+      initialization_time: Membrane.Time.monotonic_time(),
+      timestamp_base: packet.timestamp,
+      previous_timestamp: packet.timestamp,
+      clock_rate: clock_rate,
+      latency: latency
+    }
+  end
+
+  @spec complete_state_initialization_with_pad_options(
+          State.t(),
+          pad_options(),
+          RTP.payload_type(),
+          RTP.ssrc(),
+          RTP.PayloadFormat.payload_type_mapping()
+        ) :: State.t()
+  def complete_state_initialization_with_pad_options(
+        jitter_buffer_state,
+        pad_options,
+        payload_type,
+        ssrc,
+        payload_type_mapping
+      ) do
+    %{clock_rate: clock_rate} =
+      RTP.PayloadFormat.resolve(
+        payload_type: payload_type,
+        clock_rate: pad_options.clock_rate,
+        payload_type_mapping: payload_type_mapping
+      )
+
+    latency =
+      if pad_options.use_jitter_buffer do
+        time_since_initialization =
+          Time.monotonic_time() - jitter_buffer_state.initialization_time
+
+        initial_latency_left = pad_options.jitter_buffer_latency - time_since_initialization
+
+        if initial_latency_left > 0 do
+          Process.send_after(
+            self(),
+            {:initial_latency_passed, ssrc},
+            Membrane.Time.as_milliseconds(initial_latency_left, :round)
+          )
+        end
+
+        pad_options.jitter_buffer_latency
+      else
+        0
+      end
+
+    %State{jitter_buffer_state | clock_rate: clock_rate, latency: latency}
+  end
+
+  @spec insert_buffer(State.t(), Membrane.Buffer.t(), Demuxer.stream_phase()) :: State.t()
+  def insert_buffer(jitter_buffer_state, buffer, stream_phase) do
+    case stream_phase do
+      :timed_out ->
+        jitter_buffer_state
+
+      phase when phase in [:waiting_for_matching_pad, :matched_with_pad] ->
+        case BufferStore.insert_buffer(jitter_buffer_state.buffer_store, buffer) do
+          {:ok, buffer_store} ->
+            %State{jitter_buffer_state | buffer_store: buffer_store}
+
+          {:error, :late_packet} ->
+            Membrane.Logger.debug("Late packet has arrived")
+            jitter_buffer_state
+        end
+    end
+  end
+
+  @spec get_actions(State.t(), Demuxer.stream_phase(), RTP.ssrc(), Membrane.Pad.ref()) ::
+          {[Membrane.Element.Action.t()], State.t()}
+  def get_actions(jitter_buffer_state, stream_phase, ssrc, pad) do
```

Comment:
This name is quite generic, maybe something like
```suggestion
  def take_output(jitter_buffer_state, stream_phase, ssrc, pad) do
```